backend/internal/usecase/transaction.go

```diff
@@ -145,3 +167,117 @@ func (uc *TransactionUsecase) QueryIncludedTransactions(ctx context.Context, txL
 	}
 	return txs, nil
 }
+
+func (uc *TransactionUsecase) QueryTransactionDetail(ctx context.Context, txHash string) (*QueryTransactionDetailResp, *error.Http) {
+	txHashBytes, err := hex.DecodeString(txHash)
+	if err != nil {
+		err := error.NewHttpError(
+			"unable to decode tx hash",
+			"valid txHash query parameter is required",
+			http.StatusBadRequest,
+		)
+		return nil, &err
+	}
+
+	erpcTX, erpcTxErr := uc.erpcDBQuery.QueryFromTransactionDetail(ctx, txHash)
+	if erpcTxErr != nil && erpcTxErr != pgx.ErrNoRows {
+		log.Err(err).Msg("err encountered while querying erpc DB")
+		err := error.NewHttpError(
+			"error encountered while querying for data",
+			"",
+			http.StatusInternalServerError,
+		)
+		return nil, &err
+	}
+
+	tse, err := uc.observerDBQuery.QueryDecryptedTXFromSubmittedEvent(ctx, txHashBytes)
+	if err != nil {
+		log.Err(err).Msg("err encountered while querying observer DB")
+		if err == pgx.ErrNoRows {
+			err := error.NewHttpError(
+				"transaction for provided tx hash not found",
+				"",
+				http.StatusNotFound,
+			)
+			return nil, &err
+		}
+		err := error.NewHttpError(
+			"error encountered while querying for data",
+			"",
+			http.StatusInternalServerError,
+		)
+		return nil, &err
+	}
+
+	var effectiveInclusionTime int64
+	var inclusionSlot int64
+	txStatus := Pending
+	if tse.TxStatus.Valid {
+		if tse.TxStatus.TxStatusVal == data.TxStatusValIncluded {
+			txStatus = Completed
+			effectiveInclusionTime = int64(tse.DecryptedTxCreatedAtUnix)
+			if tse.Slot.Valid {
+				inclusionSlot = tse.Slot.Int64
+			}
+		} else if tse.TxStatus.TxStatusVal == data.TxStatusValNotincluded {
+			txStatus = DecryptionKeyGenerated
+		}
+	}
+
+	totalRegisteredValidators, err := uc.observerDBQuery.QueryTotalRegisteredValidators(ctx)
```

Comment:
nitpick: this will create different values over time. The estimated inclusion time should be fixed for when the transaction was sent.